From 94401378d10ff3b54172f7cedc5bc69569d2eded Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Wed, 14 Apr 2021 09:30:31 +0800
Subject: [PATCH] ipc-shm: os: modify tasklet declaration to compatible with
 kernel v5.10

In kernel v5.10, the old DECLARE_TASKLET(name, func, para) has been
droped, so correct the related codes to compatible with it.

Upstream-Status: wrlinux specific

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 os/ipc-os.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/os/ipc-os.c b/os/ipc-os.c
index 19f7026..63cf935 100644
--- a/os/ipc-os.c
+++ b/os/ipc-os.c
@@ -48,12 +48,13 @@ struct ipc_os_priv {
 static struct ipc_os_priv priv;
 
 static void ipc_shm_softirq(unsigned long arg);
-static DECLARE_TASKLET(ipc_shm_rx_tasklet, ipc_shm_softirq, IPC_SOFTIRQ_BUDGET);
+static DECLARE_TASKLET_OLD(ipc_shm_rx_tasklet, ipc_shm_softirq);
 
 /* sotfirq routine for deferred interrupt handling */
 static void ipc_shm_softirq(unsigned long budget)
 {
 	int work = 0;
+	budget = IPC_SOFTIRQ_BUDGET;
 
 	work = priv.rx_cb(budget);
 
-- 
2.25.1

